### PR TITLE
Create unit tests for postgres helpers

### LINKED ISSUE #132: Create unit test for `pgmeta.IsPK`
Add unit tests for the pgmeta methods as well as any helper function in the internal/storage/postgres.go file.

### PR DESCRIPTION
**Pull request by amoghrajesh**
Resolves #132 

- Added tests for isPk method by mocking the required framework

### CODE REVIEW

**Comment by amoghrajesh** [comment]
@prestonvasquez can you help with a review here?

File: internal/storage/postgres_test.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		},
+	}
+
+	ctx := context.Background()
+	pdb, err := NewPostgres(ctx, "postgres://postgres:localhost:5432")
+	if err != nil {
+		t.Fatalf("failed to create postgres client: %v", err)
+	}
+	pdb.meta = &dummyPgMeta
+	_, err = pdb.StartTx(ctx)
```
Comment (prestonvasquez) [inline internal/storage/postgres_test.go:34]: We don't need to start a transaction here.

File: internal/storage/postgres_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package storage
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestPGMeta(t *testing.T) {
+	dummyPgMeta := pgmeta{
```
Comment (prestonvasquez) [inline internal/storage/postgres_test.go:10]: Can we call this `mockPgMeta`?

**Review by prestonvasquez — CHANGES_REQUESTED** [review]
Hi! It's looking great! I have some initial feedback.

File: internal/storage/postgres_test.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		},
+		{
+			tableName: "table3",
+			column:    "address",
+			isPk:      true,
+		},
+	}
+
+	for _, test := range tests {
+		actual := pdb.meta.isPK(test.tableName, test.column)
```
[the remark this answers is not in the document]
Comment (amoghrajesh) [reply internal/storage/postgres_test.go:93]: Thanks. Made the changes

File: internal/storage/postgres_test.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		assert.Equal(t, actual, test.isPk)
```
[the remark this answers is not in the document]
Comment (amoghrajesh) [reply internal/storage/postgres_test.go:94]: Good idea. Made these changes too

File: internal/storage/postgres_test.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
+package storage
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestPGMeta(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (amoghrajesh) [reply internal/storage/postgres_test.go:9]: Sure. Made the change

File: internal/storage/postgres_test.go (hunk 1 above)
Comment (amoghrajesh) [reply internal/storage/postgres_test.go:34]: Removed it.

File: internal/storage/postgres_test.go [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+		},
+		bytes: map[string]int64{
+			"table1": 1234,
+			"table2": 2345,
+			"table3": 10000,
+		},
+	}
+
+	ctx := context.Background()
+	pdb, err := NewPostgres(ctx, "postgres://postgres:localhost:5432")
```
[the remark this answers is not in the document]
Comment (amoghrajesh) [reply internal/storage/postgres_test.go:29]: Made these changes

File: internal/storage/postgres_test.go (hunk 2 above)
Comment (amoghrajesh) [reply internal/storage/postgres_test.go:10]: Yeah why not. Pushing the changes now

**Comment by amoghrajesh** [comment]
@prestonvasquez just pushed my changes. Have a look again :)

File: go.mod [hunk 7]
```diff
@@ -7,30 +7,33 @@ require (
 	github.com/lib/pq v1.10.6
 	github.com/sirupsen/logrus v1.9.0
 	github.com/spf13/cobra v1.5.0
+	github.com/stretchr/testify v1.8.0
```
Comment (prestonvasquez) [inline go.mod:10]: Nice, the code looks great! Can you run `go mod tidy` locally and push the changes to undo adding the testify package?

**Review by prestonvasquez — CHANGES_REQUESTED** [review]

**Comment by prestonvasquez** [comment]
@amoghrajesh To fix the linting errors

1. Merge the main branch into your existing branch to fix the repository examples
2. To run the linter locally use `make lint`, you may have to install https://golangci-lint.run/usage/install/#local-installation
3. To run the fumpt locally use `make fmt`, you may have to install https://github.com/mvdan/gofumpt

**Comment by prestonvasquez** [comment]
@amoghrajesh I just pushed a fix for the repository example linting errors.

**Comment by amoghrajesh** [comment]
@prestonvasquez I just rebased with the master. Let us see if it works now

**Comment by amoghrajesh** [comment]
Btw, I already have the golangci-lint installed. I see this error:
```
➜  gidari git:(issue-132) ✗ make fmt 
gofumpt -l -w .
internal/storage/postgres_test.go
golangci-lint run --fix
ERRO Running error: unknown linters: 'exhaustruct,nosnakecase', run 'golangci-lint help linters' to see the list of supported linters 
```

**Comment by prestonvasquez** [comment]
@amoghrajesh 🤔 I'm not sure what could be cause that, you might need a newer version. Can you fix the non-fumpt linting issues?

**Comment by amoghrajesh** [comment]
@prestonvasquez no worries. Updated my golangci-lint. Now it should work imo

**Comment by prestonvasquez** [comment]
@amoghrajesh Looks great, thank you for the contribution!

**Comment by amoghrajesh** [comment]
Thank you @prestonvasquez!
Can you add the hacktoberfest-accepted label too?

**Comment by prestonvasquez** [comment]
@amoghrajesh It was added!

**Pull request merged**